src/Renderer/ThreeExtended/OBB.js

```diff
@@ -103,119 +105,28 @@ for (let i = 0; i < 9; i++) {
- * @param {number} centerLongitude - the longitude at the center of the portion
- * @param {boolean} isEllipsoid - should be true when computing for the globe, false otherwise
- * @param {Vector3} geodesicNormal - geodesic normal at extent's center
- * @param {number} minHeight
- * @param {number} maxHeight
- * @return {OBB}
- */
-OBB.cardinalsXYZToOBB = function cardinalsXYZToOBB(cardinals, centerLongitude, isEllipsoid, geodesicNormal, minHeight = 0, maxHeight = 0) {
-    tmp.maxV.set(-1000, -1000, -1000);
-    tmp.minV.set(1000, 1000, 1000);
-
-    let halfMaxHeight = 0;
-    tmp.normal.copy(geodesicNormal);
-    tmp.tangentPlaneAtOrigin.set(tmp.normal, 0);
-
-    // Compute the rotation transforming the tile so that it's normal becomes (0, 0, 1)
-    tmp.transformNormalToZ.setFromUnitVectors(tmp.normal, tmp.zUp);
-    // Compute the rotation to get the line [1,8,5] aligned on (0, 1, 0)
-    tmp.alignTileOnWorldXY.setFromAxisAngle(tmp.zUp, -centerLongitude);
-    const rotateTile = tmp.alignTileOnWorldXY.multiply(tmp.transformNormalToZ);
-
-    let point5InPlaneX;
-    for (let i = 0; i < cardinals.length; i++) {
-        const vec = tmp.tangentPlaneAtOrigin.projectPoint(cardinals[i], tmp.cardinal3D);
-        const d = vec.distanceTo(cardinals[i].sub(cardinals[8]));
-        halfMaxHeight = Math.max(halfMaxHeight, d * 0.5);
-        vec.applyQuaternion(rotateTile);
-        // compute tile's min/max
-        tmp.maxV.max(vec);
-        tmp.minV.min(vec);
-
-        if (i == 5) {
-            point5InPlaneX = vec.x;
-        }
-    }
-
-    const halfLength = Math.abs(tmp.maxV.y - tmp.minV.y) * 0.5;
-    const halfWidth = Math.abs(tmp.maxV.x - tmp.minV.x) * 0.5;
-
-    const nHalfSize = Math.abs(-halfMaxHeight + (minHeight - maxHeight) * 0.5);
-
-    const max = new THREE.Vector3(halfLength, halfWidth, nHalfSize);
-    const min = new THREE.Vector3(-halfLength, -halfWidth, -nHalfSize);
-
-    // delta is the distance between line `([6],[4])` and the point `[5]`
-    // These points [6],[5],[4] aren't aligned because of the ellipsoid shape
-    const delta = isEllipsoid ? (halfWidth - Math.abs(point5InPlaneX)) : 0;
-    tmp.translate.set(0, delta, -halfMaxHeight);
-
-    const obb = new OBB(min, max);
-
-    obb.lookAt(tmp.normal);
-    obb.translateX(tmp.translate.x);
-    obb.translateY(tmp.translate.y);
-    obb.translateZ(tmp.translate.z);
-
-    // for 3D
-    if (minHeight !== 0 || maxHeight !== 0) {
-        obb.translateZ(-halfMaxHeight + minHeight + nHalfSize);
-    }
+    const { sharableExtent, quaternion, position } = builder.computeSharableExtent(extent);
+    const paramsGeometry = {
+        extent: sharableExtent,
+        level: 0,
+        segment: 2,
+        disableSkirt: true,
+    };
+
+    const geometry = new TileGeometry(paramsGeometry, builder);
+    const obb = geometry.OBB;
+    obb.updateZ(minHeight, maxHeight);
+    obb.position.copy(position);
+    obb.quaternion.copy(quaternion);
+    obb.update();
```

Comment:
`geometry.dispose()` must be called before leaving the function to allow threejs to release the associated memory.

Reply:
I'm not sure because geometry was not pushed in rendering

Reply:
Yes it's useless for now.
But it doesn't hurt to add it in case threejs change its behavior and begin reserving some memory at creation time.

(or don't add it but add a comment stating something like "Calling geometry.dispose() is not needed since this geometry never gets rendered")

Reply:
comment added